frontend/src/metabase/hoc/QuestionHandler.jsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import { connect } from "react-redux";
+import { withRouter } from "react-router";
+
+import { getQuestion } from "metabase/questions/selectors";
+
+import {
+  fetchQuestion,
+  genUrlBasedQuestion,
+} from "metabase/questions/questions";
+
+const mapStateToProps = state => ({
+  question: getQuestion(state),
+  result: state.questions.currentQuestionResults && state.questions.currentQuestionResults
+});
+
+const getURL = location => location.pathname + location.search + location.hash;
+
+/*
+ * Question handler to handle fetching and URL updates when working with questions for
+ * a particular route
+ */
+class QuestionHandler extends React.Component {
+
+  static defaultProps = {
+    fetchResult: true
+  }
+
+  componentWillMount() {
+    const { params, location } = this.props;
+
+    // if we're working with a card ID,
+    // load up the question from the API
+    if (params.cardId) {
+      this._fetchSavedQuestion();
+    } else if (location.hash) {
+      // we're working with a URL based query that's in the URL hash
+      // generate a new Question object
+      this._fetchUrlQuestion();
+    }
+  }
+
+  // for now just hack this so that we fully refresh the page
+  componentWillReceiveProps(nextProps) {
```

Comment:
@tlrobinson I had a little bit of trouble getting the URL example you gave me working so I hacked this together to just force reload the page for now so I could keep going.